REFERENCE.md

```diff
@@ -0,0 +1,277 @@
+
+**Entity URI**: defined a URI to look up an existing table entity in a remote metadata registry such as Dataplex.
+
+```yaml
+rule_bindings:
+ TRANSACTIONS_UNIQUE:
+   entity_uri: dataplex://projects/<project-id>/locations/<region-id>/lakes/<lake-id>/zones/<zone-id>/entities/<entity-id> # replace variables in <angle brackets> with your own configs
+   column_id: id
+   row_filter_id: NONE
+   rule_ids:
+     - NO_DUPLICATES_IN_COLUMN_GROUPS:
+         column_names: "id"
+```
+Instead of manually specifying the `entities` configuration to be referenced in the `entity_id` field of each `Rule Binding`, you can specify an `entity_uri` containing the lookup path to a remote metadata registry referencing the table entity to be validated. This omit the need to define the `entities` configurations directly.
+
+`entity_uri` currently supports looking up BigQuery tables using Dataplex Metadata API. The BigQuery dataset containing this table must be registered as a Dataplex Asset and Dataplex Discovery Jobs must have already discovered the table and created a corresponding Dataplex `entity_id`. The Dataplex `entity_id` path can then be referenced with the URI scheme `dataplex://` in the field `entity_uri` inside a `Rule Binding` for `CloudDQ` to automatically look-up the `entity_id` using Dataplex Metadata API to resolve to the table name to be validated.
+
+#### Metadata Registry Defaults
+
+**Metadata Registry Defaults**: define default `entity_uri` values for each scheme.
+
+```yaml
+metadata_registry_defaults:
+ dataplex: # replace variables in <angle brackets> with your own configs
+   projects: <project-id> 
+   locations: <region-id> 
+   lakes: <lake-id>
+   zones: <zone-id>
+
+rule_bindings:
+ TRANSACTIONS_UNIQUE:
+   entity_uri: dataplex://entities/<entity-id> # omitting projects/locations/lakes from uri path to use the default values specified in metadata_registry_defaults
+   column_id: id
+   row_filter_id: NONE
+   rule_ids:
+     - NO_DUPLICATES_IN_COLUMN_GROUPS:
+         column_names: "id"
+
+ TRANSACTION_AMOUNT_VALID:
+   entity_uri: dataplex://zones/<override-zone-id>/entities/<entity-id> # override the default zones values specified in metadata_registry_defaults
+   column_id: amount
+   row_filter_id: NONE
+   rule_ids:
+     - VALUE_ZERO_OR_POSITIVE
+```
+Instead of providing the full URI path for every `Rule Binding`, you can define the default values you would like to use in each `entity_uri` scheme (e.g. `dataplex://`) under the `metadata_registry_defaults:` YAML node.
+
+Once `metadata_registry_defaults` is defined, you can omit the corresponding variable in the `entity_uri` to use the default values specified when resolving each `entity_uri` to a table name. You can additionally override the default by providing the variable directly in the `entity_uri`.
+
+
+
+## Library Reference
```

Comment:
should this be removed?

Reply:
The idea is that the reference guide contains two main sections: One the reference for how rules, rule bindings, row filters, and entities interact, and one the "library reference" with a "rule by rule spec". I'm not sure if it's really needed.

Reply:
I am removing the headers of sections that we haven't written yet, so that we can merge this documentation as "valid documents" even though we are planning to add more sections.